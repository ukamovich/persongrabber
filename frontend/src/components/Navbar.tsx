import React from 'react';
import {
Nav,
NavLink,
Bars,
NavMenu,
} from './NavbarElements';

const Navbar = () => {
return (
	<>
	<Nav>
		<Bars />
		<NavMenu>
		<NavLink to='/'>
			Person-Grabber
		</NavLink>

        <NavLink to='/Addperson'>
			Add person
		</NavLink>

        <NavLink to='/Addcar'>
			Add car
		</NavLink>

	
		
		</NavMenu>
	</Nav>
	</>
);
};

export default Navbar;