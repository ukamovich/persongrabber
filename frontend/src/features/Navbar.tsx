import { AppBar, Link, Toolbar, Typography } from '@mui/material';
import { navbarTheme } from './personGrabber/styles/Themes';
import React from 'react';

import { ThemeProvider } from 'styled-components';

/**
 * Navbar element with page routing
 * @returns 
 */
const Navbar = () => {
	return (
		<ThemeProvider theme={navbarTheme}>
			<AppBar 
				position="static"
				sx= {{
					bgcolor: "#63D471",
					display: "flex",
					flexFlow: "row wrap"
				}}	
			>
				<Toolbar>
					<Typography
						variant="h6"
						noWrap
						component="div"
						sx={{ 
							flexGrow: 1, 
							display: "flex",
							flexFlow: "row wrap",
							justifyContent: "flex-start",
							alignItems: "flex-end",
							gap: "20px",
						}}
						
					>
						<Link 
							href="/"
							color="inherit" 
							underline="none"
							variant="h4"
							sx= {{
								fontWeight: "bold"
							}}
						>
							Person grabber
						</Link>
						<Link 
							href="/add-person"
							color="inherit" 
							underline="none"
						>
							Add person
						</Link>
						<Link 
							href="/add-car"
							color="inherit" 
							underline="none"
						>
							Add car
						</Link>
					</Typography>

				</Toolbar>
			</AppBar >
		</ThemeProvider>
	);
};

export default Navbar;