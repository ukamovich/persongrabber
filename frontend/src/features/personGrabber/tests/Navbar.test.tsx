import Navbar from "../../Navbar";
import { shallow } from "enzyme";
import { BrowserRouter, Link } from "react-router-dom";
import renderer from "react-test-renderer";

describe("Check if navbar renders correctly", () => {
  test("Check if renders", () => {
    shallow(<Navbar />);
  });

  test("Check if navbar contains expected elements", () => {
    const navbar = shallow(<Navbar />);
    expect(navbar.find(Link)).toHaveLength(3);
  });
});

// Snapshot testing with Jest
describe("Navbar", () => {
  it("renders properly", () => {
    const treeHeader = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      )
      .toJSON();
    expect(treeHeader).toMatchSnapshot();
  });
});
