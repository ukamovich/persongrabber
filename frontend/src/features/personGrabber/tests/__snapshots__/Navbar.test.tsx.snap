// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders properly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1nq45q-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-e2l5zf-MuiTypography-root"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": "24px",
            "fontWeight": "bold",
            "textDecoration": "none",
          }
        }
      >
        Person grabber
      </a>
      <a
        href="/add-person"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": "20px",
            "textDecoration": "none",
          }
        }
      >
        Add person
      </a>
      <a
        href="/add-car"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": "20px",
            "textDecoration": "none",
          }
        }
      >
        Add car
      </a>
    </div>
  </div>
</header>
`;
